cpp/src/parquet/encoding.cc

```diff
@@ -2355,6 +2355,83 @@ class DeltaLengthByteArrayDecoder : public DecoderImpl,
   std::shared_ptr<ResizableBuffer> buffered_data_;
 };
 
+// ----------------------------------------------------------------------
+// RLE_BOOLEAN_DECODER
+
+class RleBooleanDecoder : public DecoderImpl,  virtual public BooleanDecoder {
+
+  public:
+    explicit RleBooleanDecoder(const ColumnDescriptor* descr)
+            :DecoderImpl(descr, Encoding::RLE) {}
+
+    void SetData(int num_values, const uint8_t* data, int len) override {
+      num_values_ = num_values;
+      int32_t num_bytes = 0;
+
+      if (len < 4) {
+        throw ParquetException("Received invalid length (corrupt data page?)");
```

Comment:
How about showing the invalid length too?

Reply:
Modified error message to show invalid length.